src/app/components/game/game.component.ts

```diff
@@ -118,31 +125,68 @@ export class GameComponent implements OnInit, OnChanges {
         }
         this.drawBoard();
     };
-    private _mouseDownEventListener: Function = () => {
+    private _mouseDownEventListener: Function = (e) => {
         // condition when not on latest move
         if (this.displayedMoveIndex !== this.game.moveHistory.length) {
             return;
         }
         // when mouse is pressed down
         this.CURSOR_DATA.mouseIsDown = true;
+        // left click
+        if (e.which === 1) {
+            this.CURSOR_DATA.mouseLeftIsDown = true;
+            this.CURSOR_DATA.mouseRightIsDown = false;
+            // right click
+        } else if (e.which === 3) {
+            this.CURSOR_DATA.mouseRightIsDown = true;
+            this.CURSOR_DATA.mouseLeftIsDown = false;
+        }
         if (this.CURSOR_DATA.overSquare) {
             this.CURSOR_DATA.mouseDownOn = this.CURSOR_DATA.overSquare;
-            if (!this.isPromoting) {
-                this.CURSOR_DATA.dragging = true;
+            if (this.CURSOR_DATA.mouseRightIsDown) {
+                this.CURSOR_DATA.draggedPieceIndex = -1;
+                let unique = true;
+                const newCircle = {
+                    square: new Square(
+                        this.CURSOR_DATA.mouseDownOn.x,
+                        this.CURSOR_DATA.mouseDownOn.y
+                    ),
+                    color: '#15781B',
+                    lineWidth: 5
+                };
+                const index = this.drawnCircles.findIndex(
+                    (circle) =>
+                        circle.square.toString() === newCircle.square.toString()
+                );
+
+                // delete if circle already exists, otherwise add
+                if (index !== -1) {
+                    this.drawnCircles.splice(index, 1);
+                    unique = false;
+                }
+                if (unique) {
+                    this.drawnCircles.push(newCircle);
+                }
```

Comment:
I also moved the declaration of `unique` to the bottom above where we use it (https://github.com/JBader89/schackmatt/pull/1)